File: r/nt_toolbox/toolbox_signal/perform_thresholding.R



perform_thresholding <- function(f,M,type){
  ####
  # Only 3 types of thresholding currently implemented
  ####
  
  if (type == "largest"){
    a <- abs(as.vector(MW))
    a <- a[order(-a)]
    T <- a[M+1]
    y <- f * (abs(f) > T)
  }
  else if (type == "soft"){
    s <- abs(f) - M
    s <- (s + abs(s))/2
    y <- sign(f) * s
  }
  else if (type == "hard"){
    y <- f * (abs(f) > M)
  }
  return(y)
}

File: python/nt_toolbox/perform_thresholding.py
import numpy as np

def perform_thresholding(f,M,type):
    """
        Only 3 types of thresholding currently implemented
    """
    if type == "largest":
        a = np.sort(np.ravel(abs(f)))[::-1] #sort a 1D copy of F in descending order
        T = a[M]
        y = f*(abs(f) > T)
    elif type == "soft":
        s = abs(f) - M
        s = (s + abs(s))/2
        y = np.sign(f)*s
    elif type == "hard":
        y = f*(abs(f) > M)
    return y